### PR TITLE
junixsocket: 1.3 -> 2.3.4

### LINKED ISSUE #82889: junixsocket is awfully out of date
**Describe the bug**
currently `junixsocket-1.3` is in nixpkgs, which is from 2014
since then the project has moved to GitHub and Maven (from Google Code and Ant)

An updated `default.nix`should carry sth. like
```
  name = "junixsocket";
  version = "2.3.2";

  src = fetchurl {
    url = "https://github.com/kohlschutter/junixsocket/archive/${name}-parent-${version}.tar.gz";
    sha256 = "0xggvh2zndlfspld5g184sfg42abg0ksqyk18b8c6878dhgc42ml";
  };
```

### PR DESCRIPTION
**Pull request by wamserma**
fixes #82889 but drops Darwin support and uses jar instead of
build-from-source

###### Motivation for this change

fixes #82889

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 98130"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Does not build:

```
unpacking sources                                                                                                                                                                           unpacking source archive /nix/store/7hwq3vgvzjyfzzh97d8i0ihsy8hly11k-junixsocket-dist-2.3.2-bin.tar.gz                                                                                      source root is junixsocket-dist-2.3.2                                                                                                                                                       setting SOURCE_DATE_EPOCH to timestamp 1583728005 of file junixsocket-dist-2.3.2/junixsocket-selftest-2.3.2-jar-with-dependencies.jar                                                       patching sources                                                                                                                                                                            configuring                                                                                                                                                                                 no configure script, doing nothing
building
no Makefile, doing nothing
running tests
builder for '/nix/store/4jzxz49alanrpi4b6la2nq5gcwia5k5k-junixsocket-2.3.2.drv' failed to produce output path '/nix/store/i41rrdn5q2171wq5wmx4lc8fds6nr4xv-junixsocket-2.3.2-lib'
error: build of '/nix/store/4jzxz49alanrpi4b6la2nq5gcwia5k5k-junixsocket-2.3.2.drv' failed
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 98130` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>junixsocket</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Can you remove the commented code that is no longer used/required?

**Comment by wamserma** [comment]
Yes, thanks for reminding. Updated.

**Comment by wamserma** [comment]
@SuperSandro2000 Could you add a quick approval by review and update marvin's status?

**Comment by asbachb** [comment]
Just out oft interest should be building from source beeing prefered? Since the new sources you lonkee are using maven I wonder if it's still needed to use the binary distribution!?

**Comment by wamserma** [comment]
@asbachb I haven't done much with Java since 1.4, so I would welcome some assistance with converting this into a build-from-source. Meanwhile this is a working update.

**Comment by turion** [comment]
I would also strongly recommend first trying to build from source. Maybe darwin support could even be recovered then. junixsocket changed the build system from ant to maven, and maven support in nixpkgs is not as well-documented as it could be. That makes it harder. Still, could you try?

Here is an example maven project: https://github.com/NixOS/nixpkgs/blob/e7fe577d9803885d1191c6612b95c246cb605dde/pkgs/development/java-modules/maven-hello/default.nix With a bit of luck, junixsocket will work in a similar way and doesn't need much special tweaking. If it does, we can call in some people who have packaged with maven before.

If you just want a quick result, you can also turn this MR into one that doesn't replace `junixsocket-1.3` by `junixsocket-2.3.2`, but instead just adds a new package `junixsocket-bin-2.3.2` which takes the JAR, and leave building from source for later.

File: pkgs/development/libraries/java/junixsocket/default.nix [hunk 1]
```diff
@@ -1,42 +1,44 @@
...
-    sha256 = "0c6p8vmiv5nk8i6g1hgivnl3mpb2k3lhjjz0ss9dlirisfrxf1ym";
+    url = "https://github.com/kohlschutter/junixsocket/releases/download/${pname}-parent-${version}/${pname}-dist-${version}-bin.tar.gz";
+    sha256 = "0zmvc5ddlch1s6yd88m0wfkwpddksjlkdcg0jzn5vf6cqdds020w";
   };
+  selftestjar = "${pname}-selftest-${version}-jar-with-dependencies.jar";
+in
+stdenv.mkDerivation {
+  inherit pname;
+  inherit version;
+  inherit src;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/java/junixsocket/default.nix:14]:
```suggestion
stdenv.mkDerivation rec {
  pname = "junixsocket";
  version = "2.3.4";

   src = fetchurl {
    url = "https://github.com/kohlschutter/junixsocket/releases/download/${pname}-parent-${version}/${pname}-dist-${version}-bin.tar.gz";
    sha256 = "0zmvc5ddlch1s6yd88m0wfkwpddksjlkdcg0jzn5vf6cqdds020w";
  };
```

File: pkgs/development/libraries/java/junixsocket/default.nix [hunk 2]
```diff
@@ -1,42 +1,44 @@
-{ lib, stdenv, fetchurl, ant, jdk, junit }:
-
-stdenv.mkDerivation rec {
-  name = "junixsocket-1.3";
-
+{ lib, stdenv, fetchurl, jre, runtimeShell }:
+let
+  pname = "junixsocket";
+  version = "2.3.4";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/java/junixsocket/default.nix:4]:
```suggestion
{ lib, stdenv, fetchurl, jre, runtimeShell }:

let
```

File: pkgs/development/libraries/java/junixsocket/default.nix [hunk 3]
```diff
@@ -1,42 +1,44 @@
...
   src = fetchurl {
-    url = "https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/junixsocket/${name}-src.tar.bz2";
-    sha256 = "0c6p8vmiv5nk8i6g1hgivnl3mpb2k3lhjjz0ss9dlirisfrxf1ym";
+    url = "https://github.com/kohlschutter/junixsocket/releases/download/${pname}-parent-${version}/${pname}-dist-${version}-bin.tar.gz";
+    sha256 = "0zmvc5ddlch1s6yd88m0wfkwpddksjlkdcg0jzn5vf6cqdds020w";
   };
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/java/junixsocket/default.nix:8]:
```suggestion
```

File: pkgs/development/libraries/java/junixsocket/default.nix [hunk 4]
```diff
@@ -1,42 +1,44 @@
...
+  selftestjar = "${pname}-selftest-${version}-jar-with-dependencies.jar";
+in
+stdenv.mkDerivation {
+  inherit pname;
+  inherit version;
+  inherit src;
 
-  patches = [ ./darwin.patch ];
+  preferLocalBuild = true;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/java/junixsocket/default.nix:16]:
```suggestion
```

File: pkgs/development/libraries/java/junixsocket/default.nix [hunk 5]
```diff
@@ -1,42 +1,44 @@
...
+  outputs = [ "lib" "test" "out" ];
 
-  buildInputs = [ ant jdk junit ];
+  dontConfigure = true;
+  dontBuild = true;
 
-  preConfigure =
+  installPhase =
     ''
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/java/junixsocket/default.nix:23]:
```suggestion
  installPhase = ''
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/java/junixsocket/default.nix (hunk 2 above)
Comment (wamserma) [reply pkgs/development/libraries/java/junixsocket/default.nix:4]: Having those in the let block is a personal preference for readability.

Aside from that: You're right, but this is still WIP (I'm gonna give in another try with maven, just wanted to have the version bump first.)

**Comment by timokau** [comment]
Hi!

The marvin-mk2 bot is now [discontinued](https://github.com/timokau/marvin-mk2/issues/34#issuecomment-1100656280). I have removed the relevant tags from this PR. If you still need someone to look at it, one option would be to ask in [this discourse thread](https://discourse.nixos.org/t/prs-ready-for-review/3032).

I am posting this notice to all open PRs with the `marvin` tag. Please understand that following all of these discussions would take too much time, so I will unsubscribe from this PR unless I have already been involved in it before this message.

**Pull request closed without merging**